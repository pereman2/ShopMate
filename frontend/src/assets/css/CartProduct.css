.product-entrie {
    height: 60px;
    border-radius: 7px;
    display: flex;
    flex-flow: row;
    justify-content: space-between;
    align-items: center;
    margin-left: 20px;
    margin-right: 20px;
    margin-top: 8px;
    box-shadow: 0px 0px 2px 2.5px rgba(0.15, 0.15, 0.15, 0.15);
    cursor: pointer;
    width: 100%;
  }
  
  .entrie-info {
    display: flex;
    flex-flow: column;
    justify-content: space-around;
    min-width: 60%;
  }
  
  .entrie-topline {
    display: flex;
  }
  
  .entrie-name {
    font-weight: bold;
    font-size: 18px;
  }

  .entrie-bottomline {
    display: flex;
    justify-content: space-between;
  }
  
  .entrie-image {
    max-height: 55px;
    width: 50px;
    height: auto;
    margin-left: 15px;
  }

  .entrie-price {
    font-weight: bold;
    font-size: 30px;
    margin-right: 15px;
    margin-left: 90px;
    width: auto;
  }
  