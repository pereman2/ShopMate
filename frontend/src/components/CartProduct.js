/* eslint react/prop-types: 0 */

import '../assets/css/CartProduct.css';
import imageNotFound from '../assets/images/image_not_found.jpg';
import React, { Component } from 'react';
import { capitalize } from '../utils/Utils';
import { withTranslation } from 'react-i18next';
import { roundUp } from '../utils/Utils';

class CartProduct extends Component {
  constructor(props) {
    super(props);
  }

  getEntrieImage() {
    const image = this.props.entry.item.pictures[0];
    if (image !== undefined) {
      return image;
    } else {
      return imageNotFound;
    }
  }

  getEntrieName() {
    const name = this.props.entry.item.name;
    return capitalize(name);
  }

  getEntrieQuantity() {
    const { t } = this.props;
    return t('quantity') + this.props.entry.quantity;
  }

  getEntrieUnitPrice() {
    const product = this.props.entry.item;
    const { t } = this.props;
    if (this.props.entry.quantity > 1) {
      return roundUp(product.priceWithVat, 2) + t('unitaryText');
    }
    return '';
  }

  getEntrieTotalPrice() {
    const product = this.props.entry;
    return roundUp(product.totalPrice, 2) + ' â‚¬';
  }


  render() {
    return (
      <div className="product-entrie">
        <img className="entrie-image" src={this.getEntrieImage()}></img>
        <div className="entrie-info">
          <div className="entrie-topline">
            <div className="entrie-name">{this.getEntrieName()}</div>
          </div>
          <div className="entrie-bottomline">
            <div className="entrie-quantity">
              {this.getEntrieQuantity()}
            </div>
          </div>
        </div>
        <div className="entrie-price-box">
          <div className="entrie-price">
            {this.getEntrieTotalPrice()}
          </div>
          <div className="entrie-unit-price">
            {this.getEntrieUnitPrice()}
          </div>
        </div>
      </div>
    );
  }
}

export default withTranslation()(CartProduct);
