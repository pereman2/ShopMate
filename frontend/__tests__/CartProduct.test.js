import React from 'react';
import renderer from 'react-test-renderer';

import CartProduct from '../src/components/CartProduct'

test('unitary price', () => {
    const entry = {
        item: {
            pictures: [""],
            name: "foo",
            barcode: "xdfedfwdf"
        }
    }
    const component = renderer.create(
        <CartProduct entry={entry} key={entry.item.barcode}></CartProduct>
    );
});