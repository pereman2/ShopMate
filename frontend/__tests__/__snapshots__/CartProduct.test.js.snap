// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unitary price 1`] = `
<div
  className="product-entrie"
>
  <img
    className="entrie-image"
    src=""
  />
  <div
    className="entrie-info"
  >
    <div
      className="entrie-topline"
    >
      <div
        className="entrie-name"
      >
        Foo
      </div>
    </div>
    <div
      className="entrie-bottomline"
    >
      <div
        className="entrie-quantity"
      >
        quantity1
      </div>
    </div>
  </div>
  <div
    className="entrie-price-box"
  >
    <div
      className="entrie-price"
    >
      NaN â‚¬
    </div>
    <div
      className="entrie-unit-price"
    >
      
    </div>
  </div>
</div>
`;
